### PR TITLE
feat(gatsby-dev-cli): add compiler options to gatsby-dev

### PR DESCRIPTION
**Pull request by wardpeet**
- add compiler options to gatsby-dev
- update circleci

## Description

Add compilerOptions to gatsby-dev-cli to enable patches & COMPILER_OPTIONS env vars. This Pr also adds cron job to run v4 tests

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -627,6 +642,22 @@ workflows:
           requires:
             - bootstrap-with-experimental-react
 
+  build-test-next-major:
```
Comment (ascorbic) [inline .circleci/config.yml:645]:
```suggestion
  build-test-next-major:
    executor:
      name: node
      image: "14.17.0"
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -627,6 +642,25 @@ workflows:
...
+    executor:
+      name: node
+      image: "14.17.0"
```
Comment (wardpeet) [inline .circleci/config.yml:648]:
```suggestion
      image: "14.15.0"
```

File: .circleci/config.yml (hunk 1 above)
Comment (wardpeet) [reply .circleci/config.yml:645]: Thank you

**Comment by ascorbic** [comment]
Oops. Looks like I broke it! Sorry!

**Pull request closed without merging**